_posts/2018-04-11-sign-to-contract.md

```diff
@@ -0,0 +1,400 @@
+---
+layout: post
+title: sign-to-contract 
+subtitle: How to timestamp with zero marginal cost 
+author: leonardo.comandini
+image: "/img/secp256k1commitment.png"
+---
+
+Creating a timestamp involves writing some data on the blockchain, by including those data in a valid bitcoin transaction (e.g. with `OP_RETURN`).
+The timestamp then is the cryptographic path that goes from the timestamped object to a block header, passing through the created transaction.
+However, including those extra data causes your transaction size to increase, making the overall cost of the transaction higher. 
+
+Thanks to OpenTimestamps aggregators and calendars, 
+this issue is almost completely solved by collecting several timestamp requests and 
+fitting those inside a single transaction, funded by the calendar. 
+Consequently, if you pass through a calendar, you can timestamp completely for free.
+Yet, if you want to timestamp by yourself, you have to pay some more fees.
+
+Actually there is a way to avoid to pay those extra satoshis: a *math trick* that makes possible to *include a cryptographic commitment inside an elliptic curve point*. 
+Thanks to this gimmick every purely financial transaction can be turned into a timestamping transaction, while still serving for its original purpose and without adding a single additional byte.
+
+The post is structured as follows:
+- history of the name
+- the math trick: elliptic curve commitments
+    - *pay-to-contract*
+    - *sign-to-contract*
+    - some remarks
+- an actual implementation w\ Electrum
+    - how to run
+    - how to timestamp
+    - how to verify
+- conclusions
+
+# A bit of history
+When hearing the name *sign-to-contract*, 
+you may wonder from where the contract comes out.
+To find an answer we have to go back (in 2012) to one of the [first applications](https://arxiv.org/pdf/1212.3257.pdf) that was developed: *pay-to-contract* (*sign-to-contract* elder brother) which basically allows to embed a contract inside a receiver public key.
```

Comment:
No, pay-to-contract *commits* to a contract in the public key; it doesn't *embed* the contract.